### PR TITLE
Use node interfaces for static transform broadcaster

### LINKED ISSUE #103: Static Transform Publisher should use node interfaces
Currently the static transform publisher requires a shared pointer to a complete node in order to create the publisher. This means, that as-is, this TF publisher can't be easily used in a composable node constructor, essentially because a shared pointer to `this` can't be created when the object itself is not fully created.

The feature request here is to use the `NodeTopicsInterface` instead and call `create publisher` on this instead of the full node instance.
See here:
https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/create_publisher.hpp#L28-L47

This will also ease up the tickets regarding the lifecycle nodes https://github.com/ros2/geometry2/issues/94 and https://github.com/ros2/geometry2/issues/95 as they also provide the node topic interface.

Similar work is started in here: https://github.com/ros2/geometry2/pull/100

### PR DESCRIPTION
**Pull request by Karsten1987**
fixes ros2/geometry2#103

### CODE REVIEW

**Comment by Karsten1987** [comment]
CI (packages up to tf2_ros)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6722/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3036/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5525/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6552/)

File: tf2_ros/include/tf2_ros/static_transform_broadcaster.h [hunk 1]
```diff
@@ -54,6 +54,22 @@ class StaticTransformBroadcaster{
   TF2_ROS_PUBLIC
   StaticTransformBroadcaster(rclcpp::Node::SharedPtr node);
 
+  template<class AllocatorT = std::allocator<void>>
+  StaticTransformBroadcaster(
+    rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr node_topics_interface)
+  : node_topics_interface_(node_topics_interface)
```
Comment (Karsten1987) [inline tf2_ros/include/tf2_ros/static_transform_broadcaster.h]: I am unsure whether this class should store a copy of the node interface shared_ptr or if it's superfluous.

File: tf2_ros/src/static_transform_broadcaster.cpp [hunk 2]
```diff
@@ -38,13 +38,8 @@
...
 
 StaticTransformBroadcaster::StaticTransformBroadcaster(rclcpp::Node::SharedPtr node):
-  node_(node)
-{
-  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
-  custom_qos_profile.depth = 100;
-  // TODO(tfoote) latched equivalent
-  publisher_ = node_->create_publisher<tf2_msgs::msg::TFMessage>("/tf_static", custom_qos_profile);
-}
+  StaticTransformBroadcaster(node->get_node_topics_interface())
```
Comment (wjwwood) [inline tf2_ros/src/static_transform_broadcaster.cpp]: Please put the `:` on the line with the initialization, the previous style was not following our guidelines (I guess there's no uncrustify running on this package):

```c++
StaticTransformBroadcaster::StaticTransformBroadcaster(rclcpp::Node::SharedPtr node)
: StaticTransformBroadcaster(node->get_node_topics_interface())
{}
```

File: tf2_ros/include/tf2_ros/static_transform_broadcaster.h (hunk 1 above)
Comment (wjwwood) [reply tf2_ros/include/tf2_ros/static_transform_broadcaster.h]: If it is unused, I suppose we could drop it. I don't feel strongly about it. You don't need to keep it for any lifecycle (C++ not ROS lifecycle) reason, the publisher you create will take care of that.

**Review by wjwwood — APPROVED** [review]
lgtm, that TODO you moved could be addressed now I presume, but that's out of scope for this pr.

File: tf2_ros/src/static_transform_broadcaster.cpp (hunk 2 above)
Comment (Karsten1987) [reply tf2_ros/src/static_transform_broadcaster.cpp]: I tried to be the least invasive to style. But I guess this one is okay.

**Comment by Karsten1987** [comment]
new round of CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6725/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3039/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5528/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6555/)

I'll address the latching in a second PR, because I want to make sure to test the consequences correctly.

**Pull request merged**